src/main/java/build/buildfarm/worker/InputFetchStage.java

```diff
@@ -75,7 +72,7 @@ protected OperationContext tick(OperationContext operationContext) throws Interr
     boolean success = true;
     try {
       if (Files.exists(execDir)) {
-        Worker.removeDirectory(execDir);
+        CASFileCache.removeDirectory(execDir);
```

Comment:
I think I understand why you put the method in CASFileCache, but it is quite confusing to see CASFileCache behind an interface in most places, but then referenced directly here. Can you maybe expose it in WorkerContext instead?